docs/howto/object-storage/s3/object-lock.md

```diff
@@ -0,0 +1,276 @@
+# Object lock
+
+Object lock is a feature in S3 that enables you to lock your objects
+to prevent them from being deleted or overwritten. This feature provides
+a storage solution based on the **Write Once, Read Many (WORM)** model,
+which can be helpful when you want to maintain data integrity,
+comply with regulations, or retain data for a specific period.
+
+> Are you looking to set the expiration/retention time for your
+> objects? Then you should check our [Object expiry](expiry.md) guide.
+
+## Object lock modes
+
+Object lock offers two methods for managing object retention:
+**Retention periods** and **Legal hold**.
+
+### Retention periods
+
+Retention periods specify the length of time that an object
+remains locked and protected from being overwritten or deleted.
+When you set a retention period, the object is safeguarded for
+the specified time period. You can set the retention period in
+either days or years, with a **minimum of one day** and
+no maximum limit.
+
+In addition to retention periods, you can choose the retention mode
+that applies to your objects (either Governance or Compliance). In
+{{brand}}, the only supported retention mode is **Compliance**.
+
+Compliance mode is recommended when storing compliant data,
+as it prevents objects from being overwritten or deleted
+by any user. If you configure an object with this mode, you
+cannot shorten or change its retention period, ensuring that
+the data remains **secure** and **compliant** with regulatory
+requirements.
+
+### Legal hold
+
+Legal hold is a feature designed for situations when you are uncertain
+about the length of time you need to retain an object. It can be
+enabled/disabled for any object in a locked bucket, regardless of the
+lock configuration, object retention, or age.
+
+This feature provides the same level of protection as a retention period
+but with no fixed expiration date. Instead, a legal hold will remain in
+effect until you remove it explicitly.
+
+## Prerequisites
+
+In order to manage object lock, you must install `aws` or `mc`, and [configure it with working credentials](credentials.md).
+
+You cannot (reliably) use `s3cmd` to manage this functionality.
+
+## Enabling object lock
+
+To use the object lock feature, you must first create a new bucket.
+
+> Existing buckets cannot have object lock enabled.
+> If you require assistance to enable object lock on an existing bucket,
+> please contact our support team.
+
+=== "aws"
+    ```bash
+    aws --profile <region> \
+      --endpoint-url=https://s3-<region>.citycloud.com:8080 \
```

Comment:
The `--endpoint-url` option is no longer necessary to add here (we are now instructing users to use a plugin for managing endpoints, since #139).